### PR TITLE
Initial end-to-end tests simulating host tampering, impact on historical queries

### PR DESCRIPTION
**Pull request by eddyashton**
Part of #3692.

The current tests include truncation and removal of ledger files. I have a test that corrupts ledger files, but this fails unexpectedly - I'll investigate that separately.

I haven't yet included tests that corrupt index files. The tricky thing here is that we need to create a huge volume of transactions before the index is flushed to disk, rather than kept in-memory. I've reduced the sample app's indexing thresholds so the files are at least written, but they're not relied on until we fill all buckets. This may already be a perf hit that needs to be removed, and I think to support this properly we'll need to make the index thresholds adjustable per-node, which relies on some other work (app-specific configuration).

### CODE REVIEW

File: tests/tamper_robustness.py [hunk 1]
```diff
@@ -0,0 +1,274 @@
...
+def modify_file(src_path):
+    with tempfile.TemporaryDirectory(dir=os.path.dirname(src_path)) as tmp_dir_name:
+        dst_path = os.path.join(tmp_dir_name, os.path.basename(src_path))
+        LOG.warning(f"Temporarily modifying file {src_path} (backed up at {dst_path})")
+        shutil.copy(src_path, dst_path)
+        with open(src_path, "rb+") as f:
+            content = list(f.read())
+            size = len(content)
+            for idx in (size // 3, size // 2, int(size // 1.5)):
+                content[idx] //= 2
```
Comment (wintersteiger) [inline tests/tamper_robustness.py:129]: 0 stays 0? If the files are big enough it wouldn't make much of a difference I guess. Maybe `content[idx] ^ 0b01010101`?

**Review by wintersteiger — COMMENTED** [review]
LGTM!

**Review by jumaffre — APPROVED** [review]
The new test seems to take ~90 secs. Would it be worth running it in parallel with `recovery.run_corrupted_ledger` so that it doesn't impact the total test duration and also groups these two similar tests in the same file?

**Review by raghavthind2005 — APPROVED** [review]

**Comment by eddyashton** [comment]
Closing this PR as it has rotted, and the key work is covered by #4149.

**Pull request closed without merging**